### ISSUE TITLE
Pkg operations hang on Julia 0.4.6, Windows 10 x64

### ISSUE THREAD

**Issue by evanfields** [JuliaLang/julia#17142]
Since upgrading to Julia 0.4.6, I've been unable to complete basic package operations like `Pkg.update()`, `Pkg.resolve()`, `Pkg.add("JuMP")`, etc. Even after uninstalling Julia, deleting the .julia folder, and installing a freshly downloaded copy, same thing. E.g. after the fresh install `Pkg.add("JuMP") downloaded a few dependencies and then (by REPL output, at least) got stuck, I think on a dependency of Calculus. I came back 5 hours later and nothing had changed.

For what it's worth, on another Windows 10 x64 machine I do the same work on the same repositories and thus probably have a similar package situation, and with that machine and Julia 0.4.5 I've been running package operations with no problems.

**Comment by tkelman** [comment]
Can you check in task manager which julia and git processes are running when this happens? Under details you should be able to add a column for the full command line including all arguments.

**Comment by evanfields** [comment]
Fresh install of 0.4.6 from julialang.org with a fresh .julia folder. 15 minutes on this:

```
               _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.6 (2016-06-19 17:16 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32

julia> Pkg.add("JuMP")
INFO: Initializing package repository C:\Users\Evan\.julia\v0.4
INFO: Cloning METADATA from git://github.com/JuliaLang/METADATA.jl
INFO: Cloning cache of Calculus from git://github.com/johnmyleswhite/Calculus.jl.git
```

Running git and Julia commands:
![gitjuliacommands](https://cloud.githubusercontent.com/assets/10092265/16467195/391d10a0-3e14-11e6-8314-1be4ec715e89.PNG)

**Comment by tkelman** [comment]
That's useful, thanks. Have you tried running `git config --global url."https://".insteadOf git://` from shell mode and see if that helps matters?

**Comment by evanfields** [comment]
Things seem to work correctly on 0.4.6 with the global url flag changed. 

On the nightly 0.5.0 dev, `Pkg.add` works with or without that global configuration option.

**Comment by tkelman** [comment]
I'm not entirely sure why update-index would apparently freeze. Maybe try running the same command from outside of Julia and see if it works?

**Comment by evanfields** [comment]
So (using Git bash) I ran `git --work-tree="C:\\Users\\Evan\\.julia\\v0.4\\Calculus" --git-dir="C:\\Users\\Evan\\.julia\\v0.4\\Calculus\\.git" update-index -q --really-refresh` and likewise for JuMP, and commands completed instantly with no output (no news is good news?). 

Then back to Julia (not a clean install this time), and we've been here for some hours:
![stuckinstallingarrayviews](https://cloud.githubusercontent.com/assets/10092265/16504744/9175fefa-3ee7-11e6-9af8-7664201a1d9c.PNG)

Back to git bash, where the running git command executed without output, back to Julia, where `Pkg.add("Gadfly")` is hanging with no output and the following running processes:
![gadflyhang](https://cloud.githubusercontent.com/assets/10092265/16505365/9d431918-3eea-11e6-85ec-fbc9a52e0c2c.PNG)

```
             _
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.4.6 (2016-06-19 17:16 UTC)
 _/ |\__'_|_|_|\__'_|  |  Official http://julialang.org/ release
|__/                   |  x86_64-w64-mingw32

julia> Pkg.add("Gadfly")
ERROR: InterruptException:
 in process_events at stream.jl:731
 in wait at task.jl:360
 in wait at task.jl:286
 in wait at task.jl:108
 in sync_end at task.jl:400
 [inlined code] from task.jl:422
 in add at pkg/entry.jl:64
 in add at pkg/entry.jl:73
 in anonymous at pkg/dir.jl:31
 in cd at file.jl:32
 in cd at pkg/dir.jl:31
 in add at pkg.jl:23

julia>
```

And running those commands in git bash:
```

Evan@Evan-desktop MINGW64 ~
$ git --work-tree="C:\\Users\\Evan\\.julia\\v0.4\\MathProgBase" --git-dir="C:\\Users\\Evan\\.julia\\v0.4\\MathProgBase\\.git" rev-parse --is-bare-repository
false

Evan@Evan-desktop MINGW64 ~
$
```

**Comment by evanfields** [comment]
Update: it looks like the issue is not (necessarily) with Julia 0.4.6. The package problems definitely started right after the upgrade, but after going through the previous steps in this thread I shrugged and reverted to 0.4.5 only to find that package operations were still often hanging.
![err](https://cloud.githubusercontent.com/assets/10092265/16876940/004ddc8e-4a74-11e6-9d91-a54371aea3e3.PNG)

So something on my system that's not Julia. I've tried uninstalling and deleting everything Julia related, upgrading Git to the latest version (Julia comes with its own separate copy, right?), running everything as an administrator, etc. No luck. At some seemingly non-deterministic point in the package install process, Julia will hang indefinitely while a git command is running. Using either Julia's bundled git bash or the copy independently on my system, I can run that command* and get output without hanging.

* asterisk being if the task manager says the git command running is something like `git --work-tree=C:\Users\...` then in git bash I run `git --work-tree="C:\\Users\\...`

**Comment by tkelman** [comment]
Does task manager show any other unusual activity? Antivirus or something?

**Comment by evanfields** [comment]
Task manager shows Windows 10's built-in antivirus at about 50MB of memory and .2% CPU, which doesn't change noticeably when I exit Julia. I also tried listing the Julia install directory as an exception in the Windows Defender settings and briefly turning off active protection, neither of which seemed to make any difference. 

I've also tried uninstalling git, deleting my gitconfig, replacing the git directory inside the Julia install with a corresponding directory for the most recent version, ... all no luck. Amusingly most of my searches for phrases like `git commands hang windows` turn up this thread!

I will note that in the course of all these experiments I recently observed some new behavior. The Julia REPL (fresh install of 0.4.6, attempting to `Pkg.add("DataFrames")` claimed that it was cloning the cache of BinDeps, but the git commands running had no arguments. Maybe this has happened before, but I hadn't seen it until just now:

![noargs](https://cloud.githubusercontent.com/assets/10092265/16927997/61ab480e-4cfe-11e6-8ed3-d74eec26a5d9.PNG)

By the way, I understand the Julia github issues page is not personal tech support, so if this has crossed from "bizarre behavior on Windows" to "trying to fix Evan's problem which has nothing to do with Julia," feel free to close the issue.

**Comment by tkelman** [comment]
Other have had similar mysterious git freezes, they're very hard to debug unfortunately.

**Comment by evanfields** [comment]
Something new, for posterity...
![newhang](https://cloud.githubusercontent.com/assets/10092265/17231345/6b84de3c-54ef-11e6-8ef6-5257b28d6ba5.PNG)

**Comment by tkelman** [comment]
You never got this to happen in 0.5, did you?

**Comment by evanfields** [comment]
@tkelman When I first started having the problem a month-and-change ago, I also tried a 0.5 nightly and seemed to have similar problems there. I say seemed to have because my edits in this [comment](https://github.com/JuliaLang/julia/issues/17142#issuecomment-229494920) suggest that I did have problems with 0.5, which is congruent with my memory. But that was a long time ago during a rapid fire series of installs and uninstalls and tinkering with system settings, so I may have made a mistake.

However, since the 0.5.0-rc0 release I've been using 0.5 (0.5.0-rc0+99) without problem. Go figure. :)

In fact, yesterday I used a utility to clean out all kinds of temporary files and installed the Windows 10 anniversary update, and now 0.4.6 also seems to be working. 

So yes, moderate confidence I used to have problems even with 0.5, high confidence I am no longer having problems with 0.5.
